game.py

```diff
@@ -12,15 +12,26 @@
     '-----------'
 ]
 
-
 def snowman(snowman_word):
-    """Complete the snowman function
-    replace "pass" below with your own code
-    It should print 'Congratulations, you win!'
-    If the player wins and, 
-    'Sorry, you lose! The word was {snowman_word}' if the player loses
-    """
-    pass
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    wrong_guesses_list = []
+
+    while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES and not is_word_guessed(snowman_word, correct_letter_guess_statuses):
+        print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+        user_input = get_letter_from_user(correct_letter_guess_statuses, wrong_guesses_list)
+
+        if user_input in snowman_word:
```

Comment:
Another way that this could be done is by checking the dictionary that was constructed with `build_letter_status_dict`. There are some advantages (Time Complexity) here that we've not talked about yet but it doesn't hurt to think through how it could be done!